protocol/metrics/metrics_consumer_manager.go

```diff
@@ -5,13 +5,26 @@ import (
 	"net/http"
 	"sync"
 	"sync/atomic"
+	"time"
 
 	"github.com/lavanet/lava/utils"
 	pairingtypes "github.com/lavanet/lava/x/pairing/types"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
+type LatencyTracker struct {
+	AverageLatency time.Duration
+	TotalRequests  int
+}
+
+func (lt *LatencyTracker) AddLatency(latency time.Duration) {
+	lt.TotalRequests++
+	weight := 1.0 / float64(lt.TotalRequests)
+	// Calculate the weighted average of the current average latency and the new latency
+	lt.AverageLatency = time.Duration(float64(lt.AverageLatency)*(1-weight) + float64(latency.Milliseconds())*weight)
```

Comment:
if you turn it into milliseconds then cast back into duration it will divide the time and all the usages of averageLatenncy as a time.Duration will be incorrect, so don't use .Milliseconds for the calculation